arch/risc-v/src/common/riscv_percpu.c

```diff
@@ -86,13 +123,27 @@ void riscv_percpu_init(void)
 
 void riscv_percpu_add_hart(uintptr_t hartid)
 {
-  /* Hart IDs go from 0...4 */
+  riscv_percpu_t *percpu;
+  irqstate_t      flags;
+
+  /* Make sure we are initialized */
+
+  riscv_percpu_init();
 
-  DEBUGASSERT(hartid < HART_CNT);
+  /* Get free entry for this hart, this must not fail */
+
+  flags = spin_lock_irqsave(NULL);
```

Comment:
Do we need the global lock here (and in `riscv_percpu_init`)?

If these are called only during initialization then there won't be much to gain from a local lock, but I wanted to put this here in case you wanted to consider it.

Reply:
Currently only called during init phase.

But using a local lock should be possible as the only thing that needs protecting are the module's internal, static global variables.